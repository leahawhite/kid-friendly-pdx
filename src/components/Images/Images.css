.caption-text {
  border: 1px solid #00b8e6;
  display: block;
  margin: 0 auto;
  padding: 5px;
  text-align: left;
  width: 250px;
  height: 100px;
  border-radius: 4px;
}

.uploaded-images {
  max-height: 250px;
  max-width: 250px;
  padding: 10px;
  border: 1px dashed #00b8e6;;
  margin: 10px;
  width: 70;
}

.delete {
  position: relative;
  top: 28px;
  right: -125px;
}

.button svg, .delete svg {
  filter: drop-shadow(3px 3px 3px #999);
  transition: 0.3s;
  transform: rotate(-0.2deg)
}

.button svg:hover, .delete svg:hover {
  filter: drop-shadow(4px 4px 4px #999);
  transform: rotate(-0.6deg);
  cursor: pointer
}

.delete svg {
  filter: drop-shadow(1px 1px 1px #999)
}

.delete svg:hover {
  filter: drop-shadow(1.5px 1.5px 1.5px)
}

/* Animations */
.fadein {
  animation: fadein 2s;
}

@keyframes fadein {
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
}

