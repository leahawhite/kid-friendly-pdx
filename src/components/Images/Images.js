import React from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faTimesCircle } from '@fortawesome/free-solid-svg-icons'
import './Images.css' 

export default function Images(props) {
  
  return (
    props.images.map((image, i) =>
      <div key={i} className='fadein'>
        <div 
          onClick={() => props.removeImage(image.id)} 
          className='delete'
        >
          <FontAwesomeIcon icon={faTimesCircle} size='2x' />
        </div>
        <img className="uploaded-images" src={image.src} alt={image.id} />
        <div className="Caption">
          <textarea 
            className="caption-text" 
            placeholder="Add a caption" 
            name="caption" 
            id="caption" 
            value={image.title}
            data-index={i} 
            onChange={e => props.addCaption(e)}>
          </textarea>
        </div>
      </div>
    )
  )
}

Images.defaultProps = {
  images: [],
  addCaption: () => {},
  removeImage: () => {},
}

