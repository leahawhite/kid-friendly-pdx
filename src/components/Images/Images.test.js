import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Images from './Images';

describe('Images component', () => {
  const images = [
    {
      id: 1,
      src: 'https://lh3.googleusercontent.com/ir83SjI1lJpHK4E5vQFwD7LfyE__eKaIvBtmFf1nV5wyfq-z5LVe4u7PuQvQ7ycqsFlGMe5vw3UNRB86G3SjLckIZKHgFSkz0-oQdLAAP2UxlUwq-HqDA116RjcI3iG-lA1ftsYr9S4=w2400',
      title: `Atlas Pizza's 'Jalapeno Popper' pie`,
      place_id: 1,
      user_id: 2,
    },
    {
      id: 2,
      src: 'https://lh3.googleusercontent.com/CjUvS4WgCiSRTsUUAryiPcczqAdyaSDhDg1ZTT6OrdgjfdNiTE5Q14pwI-C2m49tSjQHMnBxDgpiZyYZkxEDE3KzI3DWLsT631ROrF1zTuJcBqEXcRN8PP5U3EcG-bmMclE_uKmWh2E=w2400',
      title: 'A rainbow over Foster Blvd',
      place_id: 1,
      user_id: 1,
    },
  ]
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Images />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders empty without props', () => {
    const wrapper = shallow(<Images />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  it('renders the images given props', () => {
    const wrapper = shallow(<Images images={images} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  it('removes image when delete button is clicked', () => {
    const mockCallBack = jest.fn();
    const button = shallow((<Images removeImage={mockCallBack}>Ok!</Images>));
      button.find('.delete').simulate('click');
      expect(mockCallBack.mock.calls.length).toEqual(1);
  })
})
