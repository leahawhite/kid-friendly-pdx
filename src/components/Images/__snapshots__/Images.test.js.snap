// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Images component renders empty without props 1`] = `null`;

exports[`Images component renders the images given props 1`] = `
Array [
  <div
    className="fadein"
    key="0"
  >
    <div
      className="delete"
      onClick={[Function]}
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              512,
              512,
              Array [],
              "f057",
              "M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z",
            ],
            "iconName": "times-circle",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size="2x"
        spin={false}
        symbol={false}
        title=""
        transform={null}
      />
    </div>
    <img
      alt={1}
      className="uploaded-images"
      src="https://lh3.googleusercontent.com/ir83SjI1lJpHK4E5vQFwD7LfyE__eKaIvBtmFf1nV5wyfq-z5LVe4u7PuQvQ7ycqsFlGMe5vw3UNRB86G3SjLckIZKHgFSkz0-oQdLAAP2UxlUwq-HqDA116RjcI3iG-lA1ftsYr9S4=w2400"
    />
    <div
      className="Caption"
    >
      <textarea
        className="caption-text"
        id="caption"
        name="caption"
        onChange={[Function]}
        placeholder="Add a caption"
        value="Atlas Pizza's 'Jalapeno Popper' pie"
      />
    </div>
  </div>,
  <div
    className="fadein"
    key="1"
  >
    <div
      className="delete"
      onClick={[Function]}
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              512,
              512,
              Array [],
              "f057",
              "M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z",
            ],
            "iconName": "times-circle",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size="2x"
        spin={false}
        symbol={false}
        title=""
        transform={null}
      />
    </div>
    <img
      alt={2}
      className="uploaded-images"
      src="https://lh3.googleusercontent.com/CjUvS4WgCiSRTsUUAryiPcczqAdyaSDhDg1ZTT6OrdgjfdNiTE5Q14pwI-C2m49tSjQHMnBxDgpiZyYZkxEDE3KzI3DWLsT631ROrF1zTuJcBqEXcRN8PP5U3EcG-bmMclE_uKmWh2E=w2400"
    />
    <div
      className="Caption"
    >
      <textarea
        className="caption-text"
        id="caption"
        name="caption"
        onChange={[Function]}
        placeholder="Add a caption"
        value="A rainbow over Foster Blvd"
      />
    </div>
  </div>,
]
`;
